packages/mobile/src/firebase/saga.ts

```diff
@@ -100,55 +100,62 @@ function celoGoldExchangeRateHistoryChannel(lastTimeUpdated: number) {
   }
 
   const now = Date.now()
+  // timestamp + 1 is used because .startAt is inclusive
+  const startAt = Math.max(lastTimeUpdated + 1, now - MAX_HISTORY_RETENTION)
 
   return eventChannel((emit: any) => {
-    const emitter = (snapshot: FirebaseDatabaseTypes.DataSnapshot) => {
+    const singleItemEmitter = (snapshot: FirebaseDatabaseTypes.DataSnapshot) => {
+      emit([snapshot.val()])
+    }
+    let cancelFunction = () => {}
+    const listenForNewElements = (newElementsStartAt: number) => {
+      cancelFunction = firebase
```

Comment:
Why is this called `cancelFunction`?

Reply:
Yeah, I was doing this wrongly. The idea was to return a function that is called when channel is canceled so you stop listening, but I was not doing it correctly, we need to call the `off` function in the Firebase node ref. Switched to the correct implementation